stan/math/rev/fun/Phi_approx.hpp

```diff
@@ -44,12 +44,30 @@ namespace math {
  * @return The corresponding unit normal cdf approximation.
  */
 inline var Phi_approx(const var& a) {
-  double av = a.vi_->val_;
-  double av_squared = av * av;
-  double av_cubed = av * av_squared;
-  double f = inv_logit(0.07056 * av_cubed + 1.5976 * av);
+  double av_squared = a.val() * a.val();
+  double f = inv_logit(0.07056 * a.val() * av_squared + 1.5976 * a.val());
   double da = f * (1 - f) * (3.0 * 0.07056 * av_squared + 1.5976);
-  return var(new precomp_v_vari(f, a.vi_, da));
+  return make_callback_var(
+      f, [a, da](auto& vi) mutable { a.adj() += vi.adj() * da; });
+}
+
+template <typename T, require_var_matrix_t<T>* = nullptr>
+inline auto Phi_approx(const T& a) {
+  arena_t<value_type_t<T>> f(a.rows(), a.cols());
+  arena_t<value_type_t<T>> da(a.rows(), a.cols());
```

Comment:
Does it make sense to compute this in the forward pass? I guess we would compute av_squared twice if we compute this in the backward pass. Is that the reason?

Reply:
yeah that's the general jist. we can compute adjoint while calculating the forward pass so it's just nice to alloc everything at once, throw it in one big loop, and then have an easy peasy reverse pass.

Reply:
Cool.